### PR TITLE
trying Github docker based testing again with new compact dockerfile

### PR DESCRIPTION
**Pull request by Shimwell**
## Proposed changes

This is a second attempt at adding Github Actions docker based Ci. This should have a better chance of working as the Dockerfile was recently changed to fix the running out of space error that had been causing issues.

## Types of changes

What types of changes does your code introduce to the Paramak?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] Pep8 applied
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by RemDelaporteMathurin** [comment]
I may have some comments/suggestions that can totally be ignored of course 😄 

1. Is this intended to replace CircleCI builds ? If not then I understand that this will double up the testing time
2. Building the docker image in Github Actions won't solve #445 in CircleCI (unless changes are made elswhere).
3. Simply running` pytest tests/ ` will have the sames issues fixed in #431. Maybe splitting these tests is a bit more complicated in Github Actions ?

How about building the Docker Image in CircleCI ? https://circleci.com/docs/2.0/building-docker-images/
There are 100Gb available so this wouldn't be an issue.

Also, there is a way of triggering some tasks (like this long build) only when pushing to specific branches (say develop and main) : https://discuss.circleci.com/t/trigger-build-when-we-push-on-only-particular-branch/32169/4

what do you think ?

**Comment by Shimwell** [comment]
I shall see if splitting these tests is possible in Github Actions.

I didn't actually know circle ci can build up images so I can look into that as well.

Mainly just happy that the image is building via the other docker publish actions at the moment

**Comment by RemDelaporteMathurin** [comment]
> I shall see if splitting these tests is possible in Github Actions.
> 
> I didn't actually know circle ci can build up images so I can look into that as well.

CircleCI can build the image, test, and deploy to docker Hub if and only if the tests pass which is super convenient

**Comment by Shimwell** [comment]
ok I shall put this back to a draft while I look into circle CI a bit more. I'm quite keen on the rebuilding so that we are testing on recent versions of the dependencies. Thanks for keeping up to date with everything

**Comment by RemDelaporteMathurin** [comment]
> ok I shall put this back to a draft while I look into circle CI a bit more. I'm quite keen on the rebuilding so that we are testing on recent versions of the dependencies. Thanks for keeping up to date with everything

Well I don't know if rebuilding with Github action can then push to docker hub (with the approval of the tests) and therefore if the tests suite can use this new docker image. #445 will be solved by using the nightly (master) version of cadquery instead of version 2. Currently, the only way to fix this is to change the dockerfile in https://github.com/Shimwell/paramak_docker which isn't really convenient. It'd be better if this repo could build and push to dockerhub.

With CircleCI there could even be a way of deploying several version (say paramak, paramak:dependencies, paramak:neutronics,...) in one go if the tests pass and only on the master branch.

**Comment by Shimwell** [comment]
> > ok I shall put this back to a draft while I look into circle CI a bit more. I'm quite keen on the rebuilding so that we are testing on recent versions of the dependencies. Thanks for keeping up to date with everything
> 
> Well I don't know if rebuilding with Github action can then push to docker hub (with the approval of the tests) and therefore if the tests suite can use this new docker image. #445 will be solved by using the nightly (master) version of cadquery instead of version 2. Currently, the only way to fix this is to change the dockerfile in https://github.com/Shimwell/paramak_docker which isn't really convenient. It'd be better if this repo could build and push to dockerhub.
> 
> With CircleCI there could even be a way of deploying several version (say paramak, paramak:dependencies, paramak:neutronics,...) in one go if the tests pass and only on the master branch.

The github actions pushes to docker hub would only be triggered by merging into develop or into main, not by a PR. So we would merge if the tests pass and therefore images would only be pushed if tests pass

File: .github/workflows/docker_ci.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+jobs:  
+  build:     
+   runs-on: ubuntu-latest
+
+   steps:    
+   - uses: actions/checkout@v1
+   - name: Build and test with Docker
+     run: |
+       docker build --tag paramak .
+       docker run --rm paramak pytest /tests
```
Comment (Shimwell) [inline .github/workflows/docker_ci.yml:16]:
```suggestion
       docker run --rm paramak pytest /test_utils.py -v --cov=paramak --cov-report term --cov-report html:htmlcov --cov-report xml --junitxml=test-reports/junit.xml
       docker run --rm paramak pytest tests/test_Shape.py -v --cov=paramak --cov-append --cov-report term --cov-report html:htmlcov --cov-report xml --junitxml=test-reports/junit.xml
       docker run --rm paramak pytest tests/test_Reactor.py -v --cov=paramak --cov-append --cov-report term --cov-report html:htmlcov --cov-report xml --junitxml=test-reports/junit.xml
       docker run --rm paramak pytest tests/test_parametric_shapes/ -v --cov=paramak --cov-append --cov-report term --cov-report html:htmlcov --cov-report xml --junitxml=test-reports/junit.xml
       docker run --rm paramak pytest tests/test_parametric_components/ -v --cov=paramak --cov-append --cov-report term --cov-report html:htmlcov --cov-report xml --junitxml=test-reports/junit.xml
       docker run --rm paramak pytest tests/test_example_shapes.py -v --cov=paramak --cov-append --cov-report term --cov-report html:htmlcov --cov-report xml --junitxml=test-reports/junit.xml
       docker run --rm paramak pytest tests/test_example_components.py -v --cov=paramak --cov-append --cov-report term --cov-report html:htmlcov --cov-report xml --junitxml=test-reports/junit.xml
       docker run --rm paramak pytest tests/test_example_reactors.py -v --cov=paramak --cov-append --cov-report term --cov-report html:htmlcov --cov-report xml --junitxml=test-reports/junit.xml
       docker run --rm paramak  pytest tests/test_parametric_reactors/ -v --cov=paramak --cov-append --cov-report term --cov-report html:htmlcov --cov-report xml --junitxml=test-reports/junit.xml
```

**Comment by Shimwell** [comment]
superseded by #527

**Pull request closed without merging**